File: frontend/src/types/tool.ts
import { IToolCategory } from "./tool-category"

export interface ITool {
  id: number
  name: string
  display_name: string
  description: string
  type: string
  config: Record<string, any>
  parameters: Array<{
    name: string
    type: string
    description: string
    required: boolean
  }>
  category_id?: number
  category?: IToolCategory
  created_at: string
  updated_at: string
}

File: backend/app/entities/tool.go
package entities

type Tool struct {
	Base
	Name        string        `json:"name" gorm:"not null;size:255"`
	DisplayName string        `json:"display_name" gorm:"not null;size:255"`
	Description string        `json:"description" gorm:"type:text"`
	Type        string        `json:"type" gorm:"not null;size:50"`
	Config      SingleJSONB   `json:"config" gorm:"type:jsonb"`
	Parameters  JSONB         `json:"parameters" gorm:"type:jsonb"`
	CategoryID  *uint         `json:"category_id" gorm:"index"`
	Category    *ToolCategory `json:"category,omitempty" gorm:"foreignKey:CategoryID"`
	Chatbots    []Chatbot     `json:"chatbots,omitempty" gorm:"many2many:chatbot_tools;"`
}
